dashboard/app/routes/optimization/fine-tuning/route.tsx

```diff
@@ -0,0 +1,334 @@
+import { useFetcher, type MetaFunction } from "react-router";
+import { useEffect, useState } from "react";
+import {
+  type SFTFormValues,
+  SFTFormValuesSchema,
+  SFTFormValuesResolver,
+} from "./types";
+import { v7 as uuid } from "uuid";
+import type { SFTJob } from "~/utils/fine_tuning/common";
+import { models } from "./model_options";
+import { useRevalidator } from "react-router";
+import { useForm } from "react-hook-form";
+import { redirect } from "react-router";
+import { launch_sft_job } from "~/utils/fine_tuning/client";
+import type { ChatCompletionConfig } from "~/utils/config/variant";
+import { useConfig } from "~/context/config";
+import { FunctionSelector } from "./FunctionSelector";
+import { MetricSelector } from "./MetricSelector";
+import { VariantSelector } from "./VariantSelector";
+import {
+  dump_model_config,
+  get_fine_tuned_model_config,
+} from "~/utils/config/models";
+import { ModelSelector } from "./ModelSelector";
+import { AdvancedParametersAccordion } from "./AdvancedParametersAccordion";
+import { Button } from "~/components/ui/button";
+import { Textarea } from "~/components/ui/textarea";
+import { Form } from "~/components/ui/form";
+import type { Route } from "./+types/route";
+import type { Config } from "~/utils/config";
+
+export const meta: MetaFunction = () => {
+  return [
+    { title: "TensorZero Fine-Tuning Dashboard" },
+    { name: "description", content: "Fine-Tuning Optimization Dashboard" },
+  ];
+};
+
+// Mutable store mapping job IDs to their info
+export const jobStore: { [jobId: string]: SFTJob } = {};
+
+// If there is a job_id in the URL, grab it from the job store and pull it.
+export async function loader({ params }: Route.LoaderArgs) {
+  const job_id = params.job_id;
+
+  if (!job_id) {
+    return { status: "idle" };
+  }
+
+  const storedJob = jobStore[job_id];
+  if (!storedJob) {
+    throw new Response(JSON.stringify({ error: "Job not found" }), {
+      status: 404,
+    });
+  }
+  if (storedJob.status() === "completed" || storedJob.status() === "error") {
+    return {
+      status: storedJob.status(),
+      result: storedJob.result(),
+      modelProvider: storedJob.provider(),
+    };
+  }
+
+  try {
+    // Poll for updates
+    const updatedJob = await storedJob.poll();
+    jobStore[job_id] = updatedJob;
+
+    const result = updatedJob.result();
+    // TODO (Viraj, important!): fix the status here.
```

Comment:
Elaborate?

Reply:
this is gonna be the progress tracking thing I'll do today. I'll remove for now